crypto/rand/rand_unix.c

```diff
@@ -400,7 +400,8 @@ int rand_pool_init(void)
 
     for (i = 0; i < OSSL_NELEM(random_devices); i++)
         random_devices[i].fd = -1;
-    open_random_devices();
+    if (keep_random_devices_open)
```

Comment:
I have an issue with this change.
This assumes that rand_pool_keep_random_devices_open is called _before_
rand_pool_init but that cannot work, because rand_pool_keep_random_devices_open
calls either open_random_devices or rand_pool_cleanup, and those operate
on uninitialized data random_devices.  I believe it is a severe bug that
RAND_keep_random_devices_open does not call  RUN_ONCE(&rand_init, do_rand_init));

Reply:
Good catch.  This is a separate problem and I'll create a new PR to fix it -- we've not reached a decision as to including this PR or using #7437 instead.